src/fondant/compiler.py

```diff
@@ -207,3 +207,108 @@ def _generate_spec(
             "version": "3.8",
             "services": services,
         }
+
+
+class KubeFlowCompiler(Compiler):
+    """Compiler that creates a Kubeflow pipeline spec from a pipeline."""
+
+    def __init__(self):
+        self._resolve_imports()
+
+    def _resolve_imports(self):
+        """Resolve imports for the Kubeflow compiler."""
+        try:
+            import kfp
+
+            self.kfp = kfp
+        except ImportError:
+            msg = """You need to install kfp to use the Kubeflow compiler,\n
+                     you can install it with `pip install --extras kfp`"""
```

Comment:
```suggestion
                     you can install it with `pip install fondant[pipelines]`"""
```

Although we probably want to split this up for the different runners so it becomes fondant[kfp].